### PR TITLE
Add config for stale bot

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds config for stale bot (https://probot.github.io/apps/stale/) which will automatically mark and close issues and pull requests which haven't seen any activity for 90 days (issue is first marked after 90 days and if no one responses in 7 days, it's automatically closed).

### CODE REVIEW

**Comment by nmaludy** [comment]
:+1:

File: .github/stale.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# Number of days of inactivity before a stale Issue or Pull Request is closed.
+# Set to false to disable. If disabled, issues still need to be closed manually, but will remain marked as stale.
+daysUntilClose: 7
+
+# Issues or Pull Requests with these labels will never be considered stale. Set to `[]` to disable
+exemptLabels:
+  - WIP
+  - work in progress
+  - important
+  - feature-request
```
Comment (LindsayHill) [inline .github/stale.yml]: I think "feature" is the label we use here?

Maybe add "bug" too?

Comment (Kami) [reply .github/stale.yml]: You are correct.

Will fix it and add "bug" as well - agreed.

Comment (Kami) [reply .github/stale.yml]: @LindsayHill Updated the config.

**Comment by armab** [comment]
With our low activity of triaging issues in st2 repository and fixing really old issues, I think we're not prepared to play this recently popular game of mass-closing issues automatically.
I've seen this a lot in K8s, Helm and CNCF-related projects. In _majority_ of cases issues which highlighted a problem or discussed a design, solution that I was interested in were closed automatically.

I think starting closing stale PRs is a good first step.

Can we list the benefits of automatically closing older issues first? Why it's so important and how it's helpful for community?

**Comment by armab** [comment]
@blag Good points.

I agree it would be better to get it together and triage/revisit all the issues as part of the collective team effort instead of adding bots that in IMO will just swipe our dust under the rug.
Additionally, establish a well working process and responsibility about keeping our house clean in future with conscious decisions about closing/reviewing issues.

----

Just spending 5mins going through existing st2 issues proves that at this moment no magic bots can help with it, - it'll be faulty because part of the useful and valid issues will be eventually closed by these filters. 

I think some automation can help and assist, but the decision making should be human-driven as only @StackStorm/core has enough context.

Overall @Kami also raised a valid problem with this PR.

File: .github/stale.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Configuration for probot-stale - https://github.com/probot/stale
+
+# Number of days of inactivity before an Issue or Pull Request becomes stale
+daysUntilStale: 90
+
+# Number of days of inactivity before a stale Issue or Pull Request is closed.
+# Set to false to disable. If disabled, issues still need to be closed manually, but will remain marked as stale.
```
Comment (armab) [inline .github/stale.yml]: How about enabling only this setting? @blag @Kami 
Tag 90d issues as `stale`, but *don't* close them.

Additional bot comment for 90-days old issues could also bring some life to some stale Issues as Github notifications will come to all st2 contributors and maintainers.

That could be our chance to review the Issue, Re-tag or close it consciously as part of maintenance.

**Comment by Kami** [comment]
Thanks everyone for their comments.

As mentioned on Slack, this PR was here mostly as a proposal to start a discussion (I think the best way to do that is with some kind of proposal / prototype :)).

File: .github/stale.yml (hunk 2 above)
Comment (Kami) [reply .github/stale.yml]: I think that's a good compromise.

Comment (Kami) [reply .github/stale.yml]: Something like this should work - acd460f5f463bae4a154b0e9ace6b3f0a1f8cc6e. What do you think, @armab?

File: .github/stale.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# Set to true to ignore issues in a milestone (defaults to false)
+exemptMilestones: true
+
+# Label to use when marking as stale
+staleLabel: stale
+
+# Comment to post when marking as stale. Set to `false` to disable
+markComment: >
+  Thanks for contributing to this issue. As it has been 90 days since the last activity,
+  we are automatically closing the issue in 7 days. This is often because the request was
```
Comment (armab) [inline .github/stale.yml]: Need to adjust this comment as well

Comment (Kami) [reply .github/stale.yml]: I've already done this here - acd460f5f463bae4a154b0e9ace6b3f0a1f8cc6e (looks like you were looking at the out of date diff).

File: .github/stale.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# Set to false to disable. If disabled, issues still need to be closed manually, but will remain marked as stale.
+daysUntilClose: 7
+
+# Issues or Pull Requests with these labels will never be considered stale. Set to `[]` to disable
+exemptLabels:
+  - WIP
+  - work in progress
+  - important
+  - feature
+  - bug
```
[the remark this answers is not in the document]
Comment (Kami) [reply .github/stale.yml]: Updated - 93ad7c746e7abfb8a2013be6b0a2f633c599d528.

File: .github/stale.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+exemptMilestones: true
+
+# Label to use when marking as stale
+staleLabel: stale
+
+# Comment to post when marking as stale. Set to `false` to disable
+markComment: >
+  Thanks for contributing to this issue. As it has been 90 days since the last activity,
+  we are automatically marking is as stale. If this issue is not relevant or applicable
+  anymore (problem has been fixed in a new version or similar), please close the issue
```
[the remark this answers is not in the document]
Comment (Kami) [reply .github/stale.yml]: I don't think so. I think it needs to be done by one of the project maintainers.

[the remark this answers is not in the document]
Comment (armab) [reply .github/stale.yml]: So removing the `stale` tag manually if needed.
Good to know.

[the remark this answers is not in the document]
Comment (armab) [reply .github/stale.yml]: So we need to remove the `stale` tag manually if needed.
Good to know.

**Comment by Kami** [comment]
Will merge this and see how it works out for us.

If needed, we can adjust the config or remove the change all together.

**Pull request merged**